kombu/transport/redis.py

```diff
@@ -905,14 +906,18 @@ def _connparams(self, asynchronous=False):
             'socket_connect_timeout': self.socket_connect_timeout,
             'socket_keepalive': self.socket_keepalive,
             'socket_keepalive_options': self.socket_keepalive_options,
+            'health_check_interval': self.health_check_interval,
         }
 
         conn_class = self.connection_class
+
+        # If the connection class does not support the `health_check_interval`
+        # argument then remove it.
         if (
             hasattr(conn_class, '__init__') and
-            accepts_argument(conn_class.__init__, 'health_check_interval')
+            not accepts_argument(conn_class.__init__, 'health_check_interval')
         ):
-            connparams['health_check_interval'] = DEFAULT_HEALTH_CHECK_INTERVAL
+            connparams.pop('health_check_interval', None)
```

Comment:
The missing value argument for `pop` is not required, since the dictionary key always exists at this point. Of course it does not create a problem, but it may be misleading or cover a future bug.

Reply:
@georgepsarakis I was trying to follow the current pattern below here:
https://github.com/celery/kombu/pull/1145/files#diff-1fdc4d4a0fe8a800a42d1341599d8679R939-R941

Would you like me to update those places as well so we are consistent? Happy to leave it better than I found it :)

Reply:
I see! Except for `socket_connect_timeout`, [_filter_tcp_connparams](https://github.com/celery/kombu/pull/1145/files#diff-1fdc4d4a0fe8a800a42d1341599d8679R893) seems to discard the other options. I would not recomment to change the `socket_*` handling in this PR though, as it would complicate the diff and the test cases even further, so if you don't mind let's just apply this on current changes only.